### DISCUSSION TITLE
Support upgrading version in bzlmod single_version_override

### DISCUSSION THREAD

**Discussion by honnix** [renovatebot/renovate#32446] [Suggest an Idea]
### Tell us more.

#22610 added the feature supporting `single_version_override`, however there is no way to allow renovate to update `version` specified in `single_version_override`. While it is understandable users might intend to pin that version, in practice it is not always true. For example, when patching `MODULE.bazel` itself, it is often necessary to specify a version in `single_version_override`, but it does not necessarily mean users don't want to get version bump, as long as a new version doesn't invalidate the patch file (if it does bazel would fail immediately anyway).

I think it makes sense to add a config to enable version bump in this case.

**Comment by honnix** [comment]
I can no longer comment on #22610, but it seems if `single_version_override` has neither `version` or `registry`, the dep would still be ignored. Not entirely sure, but if so, it is not a correct assumption.

**Reply by honnix** [reply]
> I can no longer comment on #22610, but it seems if `single_version_override` has neither `version` or `registry`, the dep would still be ignored. Not entirely sure, but if so, it is not a correct assumption.

NVM this is not true.

**Comment by rarkins** [comment]
@honnix is this a feature you'd interested in adding yourself? If so then you could start right away. If not then I'd ask for a reproduction repo + example PR so that we can convert this to an issue

**Reply by honnix** [reply]
@rarkins I don't think I know enough to get this fixed, especially around how to add a config to overwrite the behaviour. Do you think this feature request needs a repro? It's not a bug but designed in this way as the documentation says.

**Reply by rarkins** [reply]
The reason for a reproduction repo is so that we can test against a real repo. It's good practice to test against a real repo for both features and fixes

**Reply by honnix** [reply]
I see. That makes sense. I will put up a repo and share it back here.

**Reply by honnix** [reply]
I created a repo here https://github.com/honnix/renovate-single-version-override-repro. Please let me know if anything else needed. Thank you.
